lib/spack/spack/spec.py

```diff
@@ -527,6 +527,13 @@ def __init__(self, spec_like, *dep_like, **kwargs):
             # XXX(deptype): default deptypes
             self._add_dependency(spec, ('build', 'link'))
 
+    def __getattr__(self, item):
+        """Delegate to self.package if the attribute is not in the spec"""
+        # This line is to avoid infinite recursion in case package is
+        # not present among self attributes
+        package = super(Spec, self).__getattribute__('package')
+        return getattr(package, item)
```

Comment:
Today I succeeded installing `cp2k` before pushing changes and right now I correctly installed `hdf5` on a fresh `develop` checkout... sorry I can't reproduce the problem...

The two lines are needed to forward requests for non-existing attributes from `Spec` to `Package`. It's what makes : 
```python
blas = spec['blas'].blas_libs
```
possible instead of : 
```python
blas = spec['blas'].package.blas_libs
```